### DISCUSSION TITLE
loginByApi command does not allow user to skip the UI logon form

### DISCUSSION THREAD

**Discussion by GrayedFox** [cypress-io/cypress-realworld-app#987] [General] [answered]
It seems like the [loginByApi command](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/support/commands.ts#L89) doesn't allow tests to skip the UI log on form, despite logging in successfully.

By cloning this repo and replacing lines [11-13](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui/user-settings.spec.ts#L11) in the `user-settings.spec.ts` with:

```
cy.loginByApi('Katharina_Bernier');
```

The issue manifests. Is there something more that command needs to do in order to allow integration tests to skip the UI logon form the same way [LoginByXState](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/support/commands.ts#L131) does?

**Comment by kevinold** [comment, marked as answer]
@GrayedFox `loginByApi` is a command meant for the [API tests](https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress/tests/api) only, not the UI tests.  `loginByXstate` is meant for [UI tests](https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress/tests/ui).

The `user-settings.spec.ts` is a UI test.

Refer to the [API tests](https://github.com/cypress-io/cypress-realworld-app/tree/develop/cypress/tests/api) for [examples of using `loginByApi`](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/api/api-bankaccounts.spec.ts#L26).

**Reply by GrayedFox** [reply]
Thanks for the pointer. I did skim over those examples, I think I'm just confused as to why this method wouldn't work for the UI (since the auth cookie is still set and carries over into the next requests). Taking a closer look at the other login commands used for UI tests now. Cheers
